### ISSUE TITLE
Fatal server error: xf86EnableIOPorts: failed to set IOPL for I/O (Operation not permitted)

### ISSUE THREAD

**Issue by Ali3n0id** [dnschneid/crouton#1877]
Got an issue getting into my precise chroot.

Followed this here guide:

https://docs.google.com/document/d/1Mj7r8QzONMUHBSIMHWLgmRCS-EfefeYj-3-AO9wt_00/edit

sudo startelementary

Entering /media/removable/chruzer/EleOS/chroots/precise...
Unknown username "dnsmasq" in message bus configuration file
/usr/bin/startelementary: Starting X server

_XSERVTransmkdir: Owner of /tmp/.X11-unix should be set to root

X.Org X Server 1.13.0
Release Date: 2012-09-05
X Protocol Version 11, Revision 0
Build Operating System: Linux 2.6.42-37-generic x86_64 Ubuntu
Current Operating System: Linux localhost 3.10.18 #1 SMP Tue May 26 18:03:35 PDT 2015 x86_64
Kernel command line: cros_secure console= loglevel=7 init=/sbin/init cros_secure oops=panic panic=-1 root=/dev/dm-0 rootwait ro dm_verity.error_behavior=3 dm_verity.max_bios=-1 dm_verity.dev_wait=1 dm="1 vroot none ro 1,0 2506752 verity payload=PARTUUID=df75ca52-6ca0-684c-9930-269761a7c42f/PARTNROFF=1 hashtree=PARTUUID=df75ca52-6ca0-684c-9930-269761a7c42f/PARTNROFF=1 hashstart=2506752 alg=sha1 root_hexdigest=e8d5eb56c862e952264ec4717e8398a1d7fa7aa9 salt=6c257451ffdb07295f3b1da1552915a8a174bdda04ee7f1063cb2b62a58da169" noinitrd vt.global_cursor_default=0 kern_guid=df75ca52-6ca0-684c-9930-269761a7c42f add_efi_memmap boot=local noresume noswap i915.modeset=1 tpm_tis.force=1 tpm_tis.interrupts=0 nmi_watchdog=panic,lapic  
Build Date: 05 November 2013  03:18:29PM
xorg-server 2:1.13.0-0ubuntu6.5~precise1 (For technical support please see http://www.ubuntu.com/support) 
Current version of pixman: 0.30.2
        Before reporting problems, check http://wiki.x.org
        to make sure that you have the latest version.
Markers: (--) probed, (**) from config file, (==) default setting,
        (++) from command line, (!!) notice, (II) informational,
        (WW) warning, (EE) error, (NI) not implemented, (??) unknown.
(++) Log file: "/tmp/Xorg.crouton.1.log", Time: Thu Jun 18 18:44:53 2015
(==) Using system config directory "/usr/share/X11/xorg.conf.d"
setversion 1.4 failed
Initializing built-in extension Generic Event Extension
Initializing built-in extension SHAPE
Initializing built-in extension MIT-SHM
Initializing built-in extension XInputExtension
Initializing built-in extension XTEST
Initializing built-in extension BIG-REQUESTS
Initializing built-in extension SYNC
Initializing built-in extension XKEYBOARD
Initializing built-in extension XC-MISC
Initializing built-in extension SECURITY
Initializing built-in extension XINERAMA
Initializing built-in extension XFIXES
Initializing built-in extension RENDER
Initializing built-in extension RANDR
Initializing built-in extension COMPOSITE
Initializing built-in extension DAMAGE
Initializing built-in extension MIT-SCREEN-SAVER
Initializing built-in extension DOUBLE-BUFFER
Initializing built-in extension RECORD
Initializing built-in extension DPMS
Initializing built-in extension X-Resource
Initializing built-in extension XVideo
Initializing built-in extension XVideo-MotionCompensation
Initializing built-in extension XFree86-VidModeExtension
Initializing built-in extension XFree86-DGA
Initializing built-in extension XFree86-DRI
Initializing built-in extension DRI2
Loading extension GLX

Fatal server error:
xf86EnableIOPorts: failed to set IOPL for I/O (Operation not permitted)

(EE) 
Please consult the The X.Org Foundation support 
         at http://wiki.x.org
 for help. 
(EE) Please also check the log file at "/tmp/Xorg.crouton.1.log" for additional information.
(EE) 
Server terminated with error (1). Closing log file.
/usr/bin/xinit: giving up
/usr/bin/xinit: unable to connect to X server: No such file or directory
/usr/bin/xinit: server error
Unmounting /media/removable/chruzer/EleOS/chroots/precise...

**Comment by divx118** [comment]
You should update your chroot see https://github.com/dnschneid/crouton/wiki/Common-issues-and-reporting

**Comment by Ali3n0id** [comment]
Hi DivX, I did try that:

chronos@localhost ~/Downloads $  sudo sh ~/Downloads/crouton -n precise -u
Password: 
Downloading latest crouton installer...
######################################################################## 100.0%
/usr/local/chroots/precise does not exist; cannot update.
Valid chroots:
EleOS wheezy
chronos@localhost ~/Downloads $  sudo sh ~/Downloads/crouton -n EleOS -u
/usr/local/chroots/EleOS already exists; updating it...
Unable to determine the release in /usr/local/chroots/EleOS. Please specify it with -r.
chronos@localhost ~/Downloads $  sudo sh ~/Downloads/crouton -n EleOS -u -r precise
/usr/local/chroots/EleOS already exists; updating it...
Release doesn't match! Please correct the -r option, or specify a second -u to
change the release, upgrading the chroot (dangerous).
chronos@localhost ~/Downloads $

**Comment by SpafinDoople** [comment]
Hey Alil3n0id,

Try the following:
a) Download a fresh crouton installer on the crouton repository home page.
b) Find the current release of your chroot by running the following:
    In crosh: `sudo enter-chroot chrootname`
    Then in the chroot shell: `lsb_release -a`
    What you see next to release should be correct. If not try the codename in the command below.
c) Update your chroot with the release: `sudo sh ~/Downloads/crouton -n chrootname -u -r release`.
d) If that does **not** work, then (I do not recommend this, as it says in your log) type:
    `sudo sh ~/Downloads/crouton -n EleOS -uu` or `sudo sh ~/Downloads/crouton -n EleOS -u -u`.

Hope I helped.

~**SpafinDoople**

**Comment by Ali3n0id** [comment]
Many thanks for the suggestion, appreciate it but still stuck with the same issue.

I have ln -s linked this chroot from a removable drive - not sure whether that has a part in this.

lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 12.04.5 LTS
Release:        12.04
Codename:       precise

sudo sh ~/Downloads/crouton -n EleOS -u -r precise
/usr/local/chroots/EleOS already exists; updating it...
Release doesn't match! Please correct the -r option, or specify a second -u to
change the release, upgrading the chroot (dangerous).

sudo sh ~/Downloads/crouton -n EleOS -uu -r precise
/usr/local/chroots/EleOS already exists; updating it...
WARNING: Changing the chroot release to precise.
Press Control-C to abort; upgrade will continue in 5 seconds.

sudo sh ~/Downloads/crouton -n EleOS -u -u -r precise
/usr/local/chroots/EleOS already exists; updating it...
WARNING: Changing the chroot release to precise.
Press Control-C to abort; upgrade will continue in 5 seconds.

**Comment by SpafinDoople** [comment]
That shouldn't matter.

Running the script should do that, just wait five seconds, though it will update your Ubuntu to 14.04

**Comment by Ali3n0id** [comment]
That's what I'm saying I've left it for five seconds on both parameters - same issue.

**Comment by SpafinDoople** [comment]
Hm....

In that case, I cannot fix your problem.

Sorry.

~**SpafinDoople**

**Comment by Ali3n0id** [comment]
No apology needed, thanks for the suggestion!

**Comment by SpafinDoople** [comment]
Thanks. :smile: 

It seems that @dnschneid has been on recently, stay tuned for updates from him.

~**SpafinDoople**

**Comment by dnschneid** [comment]
The elementary install scripts probably reorder /etc/apt/sources.list, which makes crouton freak out because it doesn't quite know what you have installed.  Do `sudo enter-chroot -u root`, edit /etc/apt/sources.list via your favorite command line editor to put the debian/ubuntu line items above the elementary ones, then exit and try updating your chroot.

It doesn't look like -uu is working properly though...

**Comment by dnschneid** [comment]
Actually, if you could post the contents of /etc/apt/sources.list before you fix it, that would be nice.

**Comment by Ali3n0id** [comment]
Hi yes here's the sources.list:

deb http://archive.ubuntu.com/ubuntu/ precise main restricted universe multiverse
deb-src http://archive.ubuntu.com/ubuntu/ precise main restricted universe multiverse
deb http://archive.ubuntu.com/ubuntu/ precise-updates main restricted universe multiverse
deb-src http://archive.ubuntu.com/ubuntu/ precise-updates main restricted universe multiverse
deb http://archive.ubuntu.com/ubuntu/ precise-security main restricted universe multiverse
deb-src http://archive.ubuntu.com/ubuntu/ precise-security main restricted universe multiverse

**Comment by Ali3n0id** [comment]
I've just added these two:

deb http://ppa.launchpad.net/elementary-os/stable/ubuntu precise main 
deb-src http://ppa.launchpad.net/elementary-os/stable/ubuntu precise main

**Comment by kngharv** [comment]
Dear all:

I am on Dell Chromebook 11 i3 version.  I am on trusty / ubuntu-gnome.  It worked less than a week, then, it broke.

The error message I got is identical to this one.

When I fired up "startgnome" command, I encountered this:
xf86EnableIOPorts: failed to set IOPL for I/O (Operation not permitted)

looking at the /tmp/Xorg.crouton.1.log,  I got this error:
intel(0): failed to open /sys/class/backlight/intel_backlight/brightness for backlight control:  Permission denied

is this due to the fact that Chromebook updated its OS and broke the custom X-server?   

thanks in advance


Harv

**Comment by Ali3n0id** [comment]
Any update on this issue?

**Comment by kngharv** [comment]
Dear all:

I reinstalled my crouton and the issue went away.

The reason why created the problem at first place is that I treated this Crouton as a regular Linux distribution.  The reality is that, it is not.  It is a Linux sitting on top of the kernel / device driver and other low-level subsystem layer that came with ChromeOS.  

It sounds obvious now, but it took me a while, as I treated it like a regular Linux myself.   

In my own case, for every new Ubuntu install, I go straight to xorg-edgers ppa and get the latest drivers.  And *THAT* was the reason why I destroyed my previous distribution.

So, when you guy installing and upgrading softwares, make sure low-level things are kept intact.  

albeit, what is considered as "low level" is not obvious.  For example, I just installed extra codecs, finger crossed...

For those who encountered the same issue, make sure you didn't update the video driver at first place.  This system rely on whatever came with ChromeOS to make it work.

Harv
